lib/instrument/transformer/dtx-decode.js

```diff
@@ -0,0 +1,70 @@
+import Stream from 'stream';
+import { DTX_MESSAGE_HEADER_LENGTH, DTX_MESSAGE_HEADER_MAGIC, DTXMessageHeader, DTXMessage} from '../headers';
+
+
+
+class DTXDecoder extends Stream.Transform {
+
+  constructor () {
+    super({ objectMode: true });
+    this._dtxManager = {};
+    this.buffer = Buffer.allocUnsafe(0);
+    this.cursor = 0;
+    this.header = undefined;
+  }
+
+  _transform (data, encoding, onData) {
+    this._decode(data);
+    onData();
+  }
+
+  _decode (data) {
+    // Merge packets
+    this.buffer = Buffer.concat([this.buffer.slice(this.cursor, this.buffer.length), data]);
+    this.cursor = 0;
+    while (this.cursor < this.buffer.length) {
+      const magic = this._recv(4);
+      if (!magic) { return; }
+      this.cursor -= 4;
```

Comment:
should the cursor be decreased or increased?

Reply:
Need decreased